src/extension/php_objects.c

```diff
@@ -194,12 +191,23 @@ dd_result dd_phpobj_load_env_values()
     zend_ini_entry *p;
     ZEND_HASH_FOREACH_PTR(EG(ini_directives), p)
     {
-        if (p->on_modify == _on_modify_wrapper &&
-            ZSTR_LEN(p->name) > NAME_PREFIX_LEN) {
+        if (p->on_modify == _on_modify_wrapper) {
+
+            size_t name_prefix_len = APPSEC_NAME_PREFIX_LEN;
+            const char *env_name_prefix = APPSEC_ENV_NAME_PREFIX;
+            size_t env_name_prefix_len = APPSEC_ENV_NAME_PREFIX_LEN;
+
+            if (strncmp(ZSTR_VAL(p->name), APPSEC_NAME_PREFIX,
```

Comment:
Instead of checking if this matches the appsec prefix, perhaps you should check if it matches the tracer prefix.